compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/hapticfeedback/HapticFeedback.uikit.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.hapticfeedback
+
+import platform.UIKit.UIImpactFeedbackGenerator
+import platform.UIKit.UISelectionFeedbackGenerator
+
+// TODO: minor UX improvement, add `prepare()` calls when internal APIs are likely to use HapticFeedback
+//  (e.g. pan started during the text selection) to reduce haptic feedback latency
+//  see https://developer.apple.com/documentation/uikit/uifeedbackgenerator
+internal class CupertinoHapticFeedback : HapticFeedback {
+    private val impactGenerator = UIImpactFeedbackGenerator()
+    private val selectionGenerator = UISelectionFeedbackGenerator()
+
+    override fun performHapticFeedback(hapticFeedbackType: HapticFeedbackType) {
+        when (hapticFeedbackType) {
+            HapticFeedbackType.LongPress -> impactGenerator.impactOccurred()
+            HapticFeedbackType.TextHandleMove -> selectionGenerator.selectionChanged()
```

Comment:
Just to highlight: it looks like haptics will also work for text inputs, which is not standard for iOS.

Reply:
Indeed. But it's dictated by `ui` library. I don't have a strong opinion on this, let's listen to people feedback about this change. I'd take some actions after the actual feedback.